src/esp/sim/Simulator.cpp

```diff
@@ -1076,6 +1076,59 @@ scene::SceneNode* Simulator::loadAndCreateRenderAssetInstance(
       assetInfo, creation, sceneManager_.get(), tempIDs);
 }
 
+#ifdef ESP_BUILD_WITH_VHACD
+std::string Simulator::convexHullDecomposition(
+    const std::string& filename,
+    const assets::ResourceManager::VHACDParameters& params,
+    const bool renderCHD,
+    const bool saveCHDToObj) {
+  Cr::Utility::Debug() << "VHACD PARAMS RESOLUTION: " << params.m_resolution;
+
+  // generate a unique filename
+  std::string CHDFilename =
+      Cr::Utility::Directory::splitExtension(filename).first + "CHD";
```

Comment:
```suggestion
      Cr::Utility::Directory::splitExtension(filename).first + ".CHD";
```

the period is not contained in `.first`, so you have to add it again

also maybe lowercase? some filesystems are case insensitive and this causes unnecessary friction